modules/installation-osp-dpdk-binding-vfio-pci.adoc

```diff
@@ -0,0 +1,192 @@
+                            pci=$(bindDriver $mac)
+                            pci_ret=$?
+                            if [[ "$pci_ret" -eq 0 ]]; then
+                                echo "$pci bind succesful"
+                            fi
+                        fi
+                    done
+                fi
+            }
+            
+            function bindDriver() {
+                local mac=$1
+                for file in /sys/class/net/*; do
+                    dev_mac=$(cat $file/address)
+                    if [[ "$mac" == "$dev_mac" ]]; then
+                        name=${file##*\/}
+                        bus_str=$(ethtool -i $name | grep bus)
+                        dev_t=${bus_str#*:}
+                        dev=${dev_t#[[:space:]]}
+            
+                        echo $dev
+            
+                        devlink="/sys/bus/pci/devices/$dev"
+                        syspath=$(realpath "$devlink")
+                        if [ ! -f "$syspath/driver/unbind" ]; then
+                            echo "File $syspath/driver/unbind not found"
+                            return 1
+                        fi
+                        if ! echo "$dev">"$syspath/driver/unbind"; then
+                            return 1
+                        fi
+            
+                        if [ ! -f "$syspath/driver_override" ]; then
+                            echo "File $syspath/driver_override not found"
+                            return 1
+                        fi
+                        if ! echo "vfio-pci">"$syspath/driver_override"; then
+                            return 1
+                        fi
+            
+                        if [ ! -f "/sys/bus/pci/drivers/vfio-pci/bind" ]; then
+                            echo "File /sys/bus/pci/drivers/vfio-pci/bind not found"
+                            return 1
+                        fi
+                        if ! echo "$dev">"/sys/bus/pci/drivers/vfio-pci/bind"; then
+                          return 1
+                        fi
+                        return 0
+                    fi
+                done
+                return 1
+            }
+            
+            for nwid in "$@"; do
+                parseNetwork $nwid
+            done
+        filesystem: root
+        mode: 0744
+        path: /usr/local/bin/vhostuser
+      - contents:
+          inline: |
+            ARG="be22563c-041e-44a0-9cbd-aa391b439a39,ec200105-fb85-4181-a6af-35816da6baf7" <1> 
+        filesystem: root
+        mode: 0644
+        path: /etc/vhostuser-bind.conf
+----
+<1> Replace this value with a comma-separated list of VFIO network UUIDs. 
+====
++
+On boot for machines that are part of this set, the MAC addresses of ports are translated into PCI bus IDs. The `vfio-pci` module is bound to any port that is assocated with a network that is identified by the {rh-openstack} network ID.
+
+.Verification
+
+* From a command line, enter following command:
```

Comment:
from a command line on the worker node (get node name with oc get nodes), e.g. `oc debug node/ci-nfv-m2gcq-master-0`; then `chroot /host`, then you can run `lspci -k` to check.